Alignment fixes + related generic helpers

shahor02 opened the pull request:
not fully operational yet, pushing to dev to reduce the time of switching between branches

shahor02 committed:
Add SymMatrixSolver to MathUtils

shahor02 committed:
Extra methods for propagation and mathutils needed for alignment

shahor02 committed:
developments for global alignment

alibuild commented:
Error while checking build/AliceO2/O2/o2/macOS-arm for 278df1e0e6c2b82e8e3b6e020ecba05471b79288 at 2022-10-15 01:38:
```
## sw/BUILD/O2-latest/log
/System/Volumes/Data/build/alice-ci-workdir/o2/sw/SOURCES/O2/10086/0/Detectors/Align/src/Controller.cxx:461:25: error: variable-sized object may not be initialized
/System/Volumes/Data/build/alice-ci-workdir/o2/sw/SOURCES/O2/10086/0/Detectors/Align/src/Controller.cxx:462:27: error: variable-sized object may not be initialized
/System/Volumes/Data/build/alice-ci-workdir/o2/sw/SOURCES/O2/10086/0/Detectors/Align/src/Controller.cxx:463:27: error: variable-sized object may not be initialized
/System/Volumes/Data/build/alice-ci-workdir/o2/sw/SOURCES/O2/10086/0/Detectors/Align/src/Controller.cxx:494:27: error: variable-sized object may not be initialized
/System/Volumes/Data/build/alice-ci-workdir/o2/sw/SOURCES/O2/10086/0/Detectors/Align/src/Controller.cxx:718:14: error: variable-sized object may not be initialized
/System/Volumes/Data/build/alice-ci-workdir/o2/sw/SOURCES/O2/10086/0/Detectors/Align/src/Controller.cxx:718:42: error: variable-sized object may not be initialized
/System/Volumes/Data/build/alice-ci-workdir/o2/sw/SOURCES/O2/10086/0/Detectors/Align/src/AlignmentTrack.cxx:1472:14: error: variable-sized object may not be initialized
/System/Volumes/Data/build/alice-ci-workdir/o2/sw/SOURCES/O2/10086/0/Detectors/Align/src/AlignmentTrack.cxx:1472:44: error: variable-sized object may not be initialized
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/278df1e0e6c2b82e8e3b6e020ecba05471b79288/build_AliceO2_O2_o2_macOS-arm/pretty.html).

alibuild commented:
Error while checking build/O2/o2-dataflow for 278df1e0e6c2b82e8e3b6e020ecba05471b79288 at 2022-10-15 02:06:
```
## sw/BUILD/O2-latest/log
431/436 Test #387: Detectors/Align/macro/MPRec2Mille.C ..................................................***Failed   32.39 sec
G__O2Align dictionary payload:1601:20: error: field has incomplete type 'o2::align::Millepede2Record'
G__O2Align dictionary payload:1420:28: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1421:41: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
G__O2Align dictionary payload:1423:28: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1424:41: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
G__O2Align dictionary payload:1426:27: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1427:40: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
99% tests passed, 1 tests failed out of 436
100% tests passed, 0 tests failed out of 102
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/278df1e0e6c2b82e8e3b6e020ecba05471b79288/build_O2_o2-dataflow/pretty.html).

alibuild commented:
Error while checking build/O2/o2-cs8 for 278df1e0e6c2b82e8e3b6e020ecba05471b79288 at 2022-10-15 02:29:
```
## sw/BUILD/O2-latest/log
427/459 Test #388: Detectors/Align/macro/MPRec2Mille.C ..................................................***Failed   29.33 sec
G__O2Align dictionary payload:1604:20: error: field has incomplete type 'o2::align::Millepede2Record'
G__O2Align dictionary payload:1423:28: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1424:41: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
G__O2Align dictionary payload:1426:28: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1427:41: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
G__O2Align dictionary payload:1429:27: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1430:40: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
99% tests passed, 1 tests failed out of 459
100% tests passed, 0 tests failed out of 102
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/278df1e0e6c2b82e8e3b6e020ecba05471b79288/build_O2_o2-cs8/pretty.html).

alibuild commented:
Error while checking build/AliceO2/O2/o2/macOS-arm for a588eab8db836866827ec5d6d3d2419b5f21eed9 at 2022-10-15 02:46:
```
## sw/BUILD/O2-latest/log
387/458 Test #387: Detectors/Align/macro/MPRec2Mille.C ..................................................***Failed    3.89 sec
G__O2Align dictionary payload:1604:20: error: field has incomplete type 'o2::align::Millepede2Record'
G__O2Align dictionary payload:1423:28: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1424:41: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
G__O2Align dictionary payload:1426:28: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1427:41: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
G__O2Align dictionary payload:1429:27: error: no matching member function for call to 'SetBit'
G__O2Align dictionary payload:1430:40: error: cannot initialize object parameter of type 'const TObject' with an expression of type 'const o2::align::Controller'
99% tests passed, 1 tests failed out of 458
100% tests passed, 0 tests failed out of 102
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/a588eab8db836866827ec5d6d3d2419b5f21eed9/build_AliceO2_O2_o2_macOS-arm/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for d52b46a81e393ceb86d773f793298afa483580e2 at 2022-10-15 05:42:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:729:38: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:730:38: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/d52b46a81e393ceb86d773f793298afa483580e2/build_O2_fullCI/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for cee00d5789e1d38e239d0429cb28a9ca3c9f90b4 at 2022-10-16 10:56:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:730:38: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:731:38: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:734:39: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:735:39: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/cee00d5789e1d38e239d0429cb28a9ca3c9f90b4/build_O2_fullCI/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for 274737f2dda1c585cbdb872f665edda29ec62e24 at 2022-10-17 01:16:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:730:38: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:731:38: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:734:39: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/Base/src/Propagator.cxx:735:39: error: explicit instantiation of 'propagateToAlphaX' does not refer to a function template, variable template, member function, member class, or static data member
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/274737f2dda1c585cbdb872f665edda29ec62e24/build_O2_fullCI/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for d8f42ac491c07023748d0a1a08d4e54592cd9824 at 2022-10-17 14:22:
```
## sw/BUILD/o2checkcode-latest/log
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/Common/test/testGPUsortCUDA.cu:22:10: error: 'boost/test/unit_test.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/TRDTracking/GPUTRDTracker.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstruction.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:45:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/display/GPUDisplay.cxx:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/Base/cuda/GPUReconstructionCUDAGenRTC.cu:16:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/sm_20_atomic_functions.h:89:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/sm_20_atomic_functions.h:89:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/benchmark.hip.cxx:199:35: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:8: error: unknown type name '__host__' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:27: error: expected ';' after top level declarator [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:33: error: overloaded 'operator+=' must have at least one parameter of class or enumeration type [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:44: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:53: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:85:1: error: unknown type name '__global__' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:19: error: use of undeclared identifier 'blockIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:32: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:45: error: use of undeclared identifier 'threadIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:78: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx:23:10: error: 'omp.h' file not found [clang-diagnostic-error]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/d8f42ac491c07023748d0a1a08d4e54592cd9824/build_O2_fullCI/pretty.html).

shahor02 committed:
Make TRD transformer thread-safe, optionally output local coordinates

so it can be used in the alignment code.
Also, make its apply-XOR option local to device (was workflow option)

shahor02 commented:
@bazinski Note that I've modified the Transformer files to make them usable in the alignment code

alibuild commented:
Error while checking build/O2/fullCI for a1f0d2e906590510076d5bf92c6f1006ce8c7104 at 2022-10-17 23:43:
```
## sw/BUILD/o2checkcode-latest/log
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/Common/test/testGPUsortCUDA.cu:22:10: error: 'boost/test/unit_test.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/TRDTracking/GPUTRDTracker.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstruction.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:45:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/display/GPUDisplay.cxx:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUTracking/Base/cuda/GPUReconstructionCUDAGenRTC.cu:16:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/sm_20_atomic_functions.h:89:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-11.7/include/sm_20_atomic_functions.h:89:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/benchmark.hip.cxx:199:35: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:8: error: unknown type name '__host__' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:27: error: expected ';' after top level declarator [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:33: error: overloaded 'operator+=' must have at least one parameter of class or enumeration type [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:44: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:53: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:85:1: error: unknown type name '__global__' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:19: error: use of undeclared identifier 'blockIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:32: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:45: error: use of undeclared identifier 'threadIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:78: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10086-slc8_x86-64/0/Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx:23:10: error: 'omp.h' file not found [clang-diagnostic-error]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10086/a1f0d2e906590510076d5bf92c6f1006ce8c7104/build_O2_fullCI/pretty.html).
